### ISSUE TITLE
agenda sidebar: exception after renaming section

### ISSUE THREAD

**Issue by damithc** [PowerPointLabs/powerpointlabs#742]
I renamed a section and synced. This exception must have been the cause of the weird behavior I experienced earlier.
```
System.Runtime.InteropServices.COMException
Shape (unknown member) : Object does not exist.
Stack Trace:
   at Microsoft.Office.Interop.PowerPoint.Shape.get_Name()
   at PowerPointLabs.AgendaLab.AgendaShape.Decode(Shape shape)
   at PowerPointLabs.AgendaLab.AgendaShape.<>c__DisplayClass1.<MeetsConditions>b__0(Shape shape)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source)
   at PowerPointLabs.AgendaLab.AgendaLabMain.UpdateBeamOnSlide(PowerPointSlide slide, Shape refBeamShape)
   at PowerPointLabs.AgendaLab.AgendaLabMain.SyncBeamOnSlides(List`1 targetSlides, PowerPointSlide refSlide)
   at PowerPointLabs.AgendaLab.AgendaLabMain.SyncBeamAgenda(SlideSelectionTracker slideTracker, PowerPointSlide refSlide)
   at PowerPointLabs.AgendaLab.AgendaLabMain.SynchroniseAgenda()
   at PowerPointLabs.Ribbon1.UpdateAgendaClick(IRibbonControl control)
```

**Comment by kai33** [comment]
Bug cause:
https://github.com/PowerPointLabs/powerpointlabs/blob/dev-2.0/PowerPointLabs/PowerPointLabs/AgendaLab/AgendaLabMain.cs#L882

shape.Delete() will deallocate memory for the shape, but the deleted shape still remains in the collection.

when this shape is accessed again (e.g. shape.Name) here:
https://github.com/PowerPointLabs/powerpointlabs/blob/dev-2.0/PowerPointLabs/PowerPointLabs/AgendaLab/AgendaLabMain.cs#L889

it then throws exception error. bcos the shape is deleted.

**Comment by Ohohcakester** [comment]
Fixed in 03bd8f05e7b6a5cf47198198a15602bbd3c8815c
